### PR TITLE
feat(notetaker): accept decline notetaker ai generation

### PR DESCRIPTION
**Pull request by adityanandanx**
# Description

- Ability to accept or decline the ai generated content in the notetaker.
- Notetaker Search bar is now visible by default and can be toggled via `Ctrl+F` or the `Ask Brain` Button.
- Ability to select content and ask questions with respect to the selection.

## Checklist before requesting a review

Please delete options that are not relevant.

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [ ] I have commented hard-to-understand areas
- [ ] I have ideally added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged

## Screenshots (if appropriate):
image
image

### CODE REVIEW

File: frontend/lib/components/TextEditor/TextEditor.tsx [hunk 1]
```diff
@@ -52,24 +70,79 @@ export const TextEditor = (): JSX.Element => {
...
+
+  const toggleSearchBar = () => {
+    if (searchBarOpen) {
+      setSearchBarOpen(false);
+      editor?.commands.focus();
+    } else {
+      setSearchBarOpen(true);
+      searchEditorRef.current?.commands.focus();
+    }
+  };
```
Comment (Zewed) [inline frontend/lib/components/TextEditor/TextEditor.tsx:94]:
```suggestion
const toggleSearchBar = () => {
  setSearchBarOpen(!searchBarOpen);
  (searchBarOpen ? editor : searchEditorRef.current)?.commands.focus();
};
```

File: frontend/lib/components/TextEditor/TextEditor.tsx [hunk 2]
```diff
@@ -52,24 +70,79 @@ export const TextEditor = (): JSX.Element => {
...
 
   return (
     <div className={styles.main_container}>
       <div className={styles.editor_wrapper}>
-        <Toolbar searchBarEditor={searchEditorRef.current} editor={editor} />
+        <Toolbar toggleSearchBar={toggleSearchBar} editor={editor} />
+        <div>
+          <BubbleMenu
+            shouldShow={() => {
+              return editor.isActive("aiHighlight", { type: "ai" });
```
Comment (Zewed) [inline frontend/lib/components/TextEditor/TextEditor.tsx:107]: can you store aiHighlight (variable name) in a variable, as it's used several times :)

File: frontend/lib/components/TextEditor/extensions/AIHighlight.ts [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+            return false;
+          }
+
+          tr.removeMark(pos, pos + node.nodeSize, mark.type);
+        }
+      });
+    }
+  });
+
+  dispatch(tr);
```
Comment (Zewed) [inline frontend/lib/components/TextEditor/extensions/AIHighlight.ts:26]: can you be more explicit in variable names here ? :)

Comment (Zewed) [reply frontend/lib/components/TextEditor/extensions/AIHighlight.ts:26]:
```suggestion
state.doc.descendants((node, pos) => {
  if (node.isText) {
    node.marks.some((mark) => 
      mark.type.name === "aiHighlight" && mark.attrs["type"] === "selection" &&
      tr.removeMark(pos, pos + node.nodeSize, mark.type)
    );
  }
});
```

File: frontend/lib/components/TextEditor/extensions/types.d.ts [hunk 4]
```diff
@@ -0,0 +1,33 @@
+import { HighlightOptions } from "@tiptap/extension-highlight";
+
+export type AIHighlightOptions = HighlightOptions & {
+  type: "ai" | "selection";
```
Comment (Zewed) [inline frontend/lib/components/TextEditor/extensions/types.d.ts:4]: declare a type here, and reuse it in AIHihlight.ts

**Review by Zewed — CHANGES_REQUESTED** [review]

File: frontend/lib/components/TextEditor/extensions/AIHighlight.ts (hunk 3 above)
Comment (adityanandanx) [reply frontend/lib/components/TextEditor/extensions/AIHighlight.ts:26]: This raises an eslint error
image
Is this better?
```ts
  state.doc.descendants((node, pos) => {
    if (node.isText) {
      node.marks.map((mark) => {
        if (
          mark.type.name === AIHighlight.name &&
          mark.attrs["type"] === "selection"
        ) {
          tr.removeMark(pos, pos + node.nodeSize, mark.type);
        }
      });
    }
  });

```

Comment (Zewed) [reply frontend/lib/components/TextEditor/extensions/AIHighlight.ts:26]: Yes perfect

**Pull request merged**